service/src/test/java/com/commercetools/pspadapter/tenant/TenantFactoryTest.java

```diff
@@ -0,0 +1,75 @@
+package com.commercetools.pspadapter.tenant;
+
+import com.commercetools.pspadapter.payone.config.PayoneConfig;
+import com.commercetools.pspadapter.payone.domain.ctp.CustomTypeBuilder;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TenantFactoryTest {
+
+    @Mock
+    private PayoneConfig payoneConfig;
+
+    @Mock
+    private TenantConfig tenantConfig;
+
+    private TenantFactory factory;
+
+    @Before
+    public void setUp() throws Exception {
+        when(payoneConfig.getApiUrl()).thenReturn("http://test.api.url");
+        when(tenantConfig.getPayoneConfig()).thenReturn(payoneConfig);
+        when(tenantConfig.getName()).thenReturn("testTenantName");
+
+        factory = new TenantFactory("testPayoneInterfaceName", tenantConfig);
+    }
+
+    @Test
+    public void getsTenantName() throws Exception {
+        assertThat(factory.getTenantName()).isEqualTo("testTenantName");
+    }
+
+    @Test
+    public void getsPayoneInterfaceName() throws Exception {
+        assertThat(factory.getPayoneInterfaceName()).isEqualTo("testPayoneInterfaceName");
+    }
+
+    @Test
+    public void getsPaymentHandlerUrl() throws Exception {
+        assertThat(factory.getPaymentHandlerUrl()).contains("/testTenantName/");
```

Comment:
this is a Unit test, this unit (`TenantFactory`) doesn't care about base URL - it might be anything. This exact test verifies only that `getPaymentHandlerUrl()` attaches tenant name to URL.

Testing the base URL is a purpose of integration tests.